language/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -147,19 +150,20 @@ pub trait MoveTestAdapter<'a>: Sized {
         args: Vec<<<Self as MoveTestAdapter<'a>>::ExtraValueArgs as ParsableValue>::ConcreteValue>,
         gas_budget: Option<u64>,
         extra: Self::ExtraRunArgs,
-    ) -> Result<(Option<String>, SerializedReturnValues)>;
+    ) -> Result<(Option<String>, SerializedReturnValues, Option<Value>)>;
+
     fn view_data(
         &mut self,
         address: AccountAddress,
         module: &ModuleId,
         resource: &IdentStr,
         type_args: Vec<TypeTag>,
-    ) -> Result<String>;
+    ) -> Result<(String, Value)>;
```

Comment:
Two things:
1) I think we should maybe make this always return some form of structured value, and move the rendering into the framework
2) I think the structured value should be some enum that is essentially `Value { ModuleMetaData { ... } | Resource(MoveValue) }

Reply:
I would like to use an enum to replace all function's return values, but not all functions return the same results.